code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectResourceManager.java

```diff
@@ -54,148 +56,84 @@
 
     @Override
     public Class<?>[] getTypeClasses() {
-        return new Class[] {};
+        return new Class<?>[0];
     }
 
     @Override
     public Object listInternal(SchemaFactory schemaFactory, String type, Map<Object, Object> criteria, ListOptions options) {
-        ApiRequest apiRequest = ApiContext.getContext().getApiRequest();
-        String token = getTokenFromRequest(apiRequest);
-        GithubUtils.AccesibleIds ids = githubUtils.validateAndFetchAccesibleIdsFromToken(token);
-        String id = (String) criteria.get("id");
-        if (StringUtils.isNotEmpty(id)) {
+        Policy policy = (Policy) ApiContext.getContext().getPolicy();
+        Object obj = criteria.get("id");
+        String id = "";
+        if (obj != null) {
+            if (obj instanceof String) {
+                id = (String) obj;
+            }else {
+                throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+            }
             Account account = getObjectManager().loadResource(Account.class, id);
             if (null == account) {
                 return null;
             }
-            if (!hasAccess(account, ids)) {
-                throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
+            if (!authDao.hasAccessToProject(account.getId(), policy.getAccountId(),
+                    policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())) {
+                    throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+            }
+            policy.grantObjectAccess(account);
+            return Collections.singletonList(account);
+        }
+        if (criteria.get("uuid") != null) {
+            String uuid = ((String) ((Condition) ((ArrayList) criteria.get("uuid")).get(0)).getValue());
+            Account project = authDao.getAccountByUuid(uuid);
+            if (authDao.hasAccessToProject(project.getId(), policy.getAccountId(),
+                    policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())) {
+                return project;
+            } else {
+                throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
```

Comment:
In the current code, do we ever throw not found exception if search criteria returns an empty result? From the API, I only saw empty list being returned, never an exception thrown. Please confirm with Vince.